### DISCUSSION TITLE
Duplicate instances/centroids detected leads to switching

### DISCUSSION THREAD

**Discussion by olinesn** [talmolab/sleap#807] [Help!]
Hello,

When proofreading inference, I've noticed that a substantial number of my track switches occur on frames where the same animal is detected twice. Here's an example: 

Frames 0-6859: track_01
Frame 6860: track_01, track_13 on the same animal
Frames 6861-forward: track_13

Is there a way to update flow so that it will bias toward tracks with some history? Or is it more fruitful to try to eliminate duplicate instances in my model, or by running a cleanup step before creating tracks?

![Bug_01](https://user-images.githubusercontent.com/69451171/175080575-d064ef92-7b0a-41c2-b1f4-43781da624f2.png)
![Bug_02](https://user-images.githubusercontent.com/69451171/175080577-478bf09e-a920-43e6-85aa-2f8930349dd4.png)
![Bug_03](https://user-images.githubusercontent.com/69451171/175080579-d86ebf06-24f5-4b94-a659-a07f7642c711.png)

Thanks!
Stefan

**Comment by talmo** [comment]
Hi @olinesn,

Have you tried using the [culling options with a lower IOU](https://sleap.ai/guides/proofreading.html#tracking-methods)? We can try something more custom but that should do something like what you want.

The other option is to try decreasing the sigma for top-down centroid models so there's less ambiguity about the peak location.

Let us know!

Talmo

**Reply by olinesn** [reply]
Thanks for your input Talmo. 

I tried out these settings and added "--tracking.clean_iou_threshold .8" to the command line input, which looks like it's the default listed in the gui, and get similar results, so I'll try lowering that number now. 

This input gave me the following output:
sleap-track --verbosity rich --tracking.tracker flow  --tracking.target_instance_count 2 --tracking.clean_instance_count 2 --tracking.clean_iou_threshold .8 --tracking.post_connect_single_breaks 1 --tracking.track_window 5 --tracking.similarity centroid --model "G:\My Drive\Experiments\sLEAP\Projects\Hallucination\Top_hallucination_SO\models\220610_052210.centroid.n=7246" --model "G:\My Drive\Experiments\sLEAP\Projects\Hallucination\Top_hallucination_SO\models\220610_141517.centered_instance.n=7246" "G:\Shared drives\Falkner Lab\Video_annotation\Oline_data\2022-03-02_Mouse2971\2022-04-11\top-04112022163029-0000_h265_CRF14_denoised.mp4"

![image](https://user-images.githubusercontent.com/69451171/175936749-5b6688b9-26c9-4252-8612-4edc9e7824f6.png)

I don't quite have a feel for how much to lower the clean_iou_threshold, so I'll start by cutting it in half to 0.4.

**Reply by olinesn** [reply]
P.S. if this avenue doesn't work, I'll try out tightening the centroid sigma, thanks for that idea.

**Comment by olinesn** [comment]
Actually maybe I misinterpreted, and should be using "--tracking.pre_cull_iou_threshold" ? If the default is 0, what type of manual input would make sense?

**Reply by roomrys** [reply]
Hi @olinesn,

The `--tracking.pre_cull_iou_threshold` works similarly to the `--tracking.clean_iou_threshold` option, but will eliminate overlapping instances **before** assigning tracks. This seems to be the option you would want. The higher the number the more the instances need to overlap for SLEAP to eliminate one of them. Your instances seem to be right on top of each other so I would start with a higher number such as 0.8 just so that we do not eliminate unique instances that happen to overlap some.

Let us know how it goes!

Thanks,
Liezl
